### DISCUSSION TITLE
Organize resources by feature

### DISCUSSION THREAD

**Discussion by Pacheco95** [adonisjs/core#1940] [Help]
Is there a way to organize the project app folder in a feature like way (like NestJS do)? For small projects, it's ok to use separated folders by category like controllers, models, migrations, etc. But as application grows it becomes a pain to manage resources with feature like organization.

**Comment by prescience-data** [comment]
I assume you could just modify your `aliases` in your `adonisrc.json` file?

## Old
```json
  "aliases": {
    "App": "app",
    "Config": "config",
    "Database": "database",
    "Contracts": "contracts"
  },
```

## New
```json
  "aliases": {
    "App": "app",
    "Config": "config",
    "FooFeature": "app/Features/Foo", 
    "BarFeature": "app/Features/Bar"
  },
```
Whereby `app/Features/Foo` might have:

- `app/Features/Foo`
  - `Controllers`
    - `Home.ts`
    - `About.ts`
  - `Models`
    - `Post.ts`
    - `Comment.ts`

_..and so on..._

Then accessed via 

```ts
import Post from "FooFeature/Models/Post"
```

No idea if it will work though as I haven't tested.

**Reply by Pacheco95** [reply]
Nice insight. I'll try it!

**Reply by prescience-data** [reply]
Btw you'll need to add it to `tsconfig.json` as well.
